pkg/apis/componentconfig/v1alpha1/types.go

```diff
@@ -21,16 +21,18 @@ import (
 )
 
 // ClientConnectionConfiguration contains details for constructing a client.
+// TODO: This struct should be referenced from k8s.io/apimachinery/pkg/apis/config instead.
 type ClientConnectionConfiguration struct {
 	// kubeConfigFile is the path to a kubeconfig file.
-	KubeConfigFile string `json:"kubeconfig"`
+	// TODO: This field maybe should be named just `KubeConfig`
+	KubeConfigFile string `json:"kubeConfigFile"`
```

Comment:
is there a reason you're changing the external serialization of this? I expected this instead:
```
	Kubeconfig string `json:"kubeconfig"`
```